packages/react/src/ReactElementValidator.js

```diff
@@ -177,6 +180,32 @@ function validateChildKeys(node, parentType) {
   }
 }
 
+function getPropTypes(type) {
+  const typeOfType = typeof type;
+  if (typeOfType === 'string') {
+    // Host component
+    return null;
+  } else if (typeOfType === 'function') {
+    // Functions/classes
+    return type.propTypes;
+  } else if (type !== null && typeOfType === 'object') {
+    // Special types
+    let innerType = null;
+    switch (type.$$typeof) {
+      case REACT_MEMO_TYPE:
+        innerType = type.type;
+        break;
+      case REACT_LAZY_TYPE: {
+        innerType = refineResolvedLazyComponent(type);
+        break;
+      }
+    }
+    const innerPropTypes = getPropTypes(innerType);
+    return {...innerPropTypes, ...type.propTypes};
```

Comment:
Shouldn’t this intersect if there are two checkers for the same key?

Reply:
Hmm. Right. I wasn't sure what's the correct mental model but intersecting makes sense.

Reply:
Meh. This seems too complicated. Either would have to always re-create them, or cache them somewhere. I think shadowing is reasonable, no?

Reply:
If shadowing is reasonable, why not the whole set? It should probably be a warning that the type sets are ambiguous and only one should be provided. Seems like an error.